runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillStateReader.java

```diff
@@ -472,7 +485,13 @@ public void startBatchAndBlock() {
       }
 
       Windmill.KeyedGetDataRequest request = createRequest(toFetch);
-      Windmill.KeyedGetDataResponse response = server.getStateData(computation, request);
+      Windmill.KeyedGetDataResponse response;
+      try (AutoCloseable readWrapper =
+          readWrapperSupplier == null ? null : readWrapperSupplier.get()) {
```

Comment:
create a no-op supplier in constructor so that when running it is always non-null

Reply:
Done